test/rules_std_test.py

```diff
@@ -50,57 +52,79 @@ def assert_rule_fail_in_sql(code, sql):
-    ('L004', 'fail', '   \n  \t \n  SELECT 1', None),
-    ('L005', 'fail', 'SELECT 1 ,4', 'SELECT 1,4'),
-    ('L008', 'pass', 'SELECT 1, 4', None),
-    ('L008', 'fail', 'SELECT 1,   4', 'SELECT 1, 4'),
-    ('L014', 'pass', 'SELECT a, b', None),
-    ('L014', 'pass', 'SELECT A, B', None),
-    ("L015", 'fail', 'SELECT DISTINCT(a)', None),
-    ("L015", 'fail', 'SELECT DISTINCT(a + b) * c', None),
+@pytest.mark.parametrize("rule,pass_fail,qry,fixed,configs", [
+    ("L001", 'fail', 'SELECT 1     \n', 'SELECT 1\n', None),
+    ('L002', 'fail', '    \t    \t    SELECT 1', None, None),
+    ('L003', 'fail', '     SELECT 1', '    SELECT 1', None),
+    ('L004', 'pass', '   \nSELECT 1', None, None),
+    ('L004', 'pass', '\t\tSELECT 1\n', None, None),
+    ('L004', 'fail', '   \n  \t \n  SELECT 1', None, None),
+    ('L005', 'fail', 'SELECT 1 ,4', 'SELECT 1,4', None),
+    ('L008', 'pass', 'SELECT 1, 4', None, None),
+    ('L008', 'fail', 'SELECT 1,   4', 'SELECT 1, 4', None),
+    ('L014', 'pass', 'SELECT a, b', None, None),
+    ('L014', 'pass', 'SELECT A, B', None, None),
+    # Check we get fails for using DISTINCT apparently incorrectly
+    ("L015", 'fail', 'SELECT DISTINCT(a)', None, None),
+    ("L015", 'fail', 'SELECT DISTINCT(a + b) * c', None, None),
     # Space after DISTINCT makes it okay...
-    ("L015", 'pass', 'SELECT DISTINCT (a)', None),  # A bit iffy...
-    ("L015", 'pass', 'SELECT DISTINCT (a + b) * c', None),  # Definitely okay
+    ("L015", 'pass', 'SELECT DISTINCT (a)', None, None),  # A bit iffy...
+    ("L015", 'pass', 'SELECT DISTINCT (a + b) * c', None, None),  # Definitely okay
     # Test that fixes are consistent
-    ('L014', 'fail', 'SELECT a,   B', 'SELECT a,   b'),
-    ('L014', 'fail', 'SELECT B,   a', 'SELECT B,   A'),
+    ('L014', 'fail', 'SELECT a,   B', 'SELECT a,   b', None),
+    ('L014', 'fail', 'SELECT B,   a', 'SELECT B,   A', None),
     # Test that we don't fail * operators in brackets
-    ('L006', 'pass', 'SELECT COUNT(*) FROM tbl\n', None),
+    ('L006', 'pass', 'SELECT COUNT(*) FROM tbl\n', None, None),
+    # Long lines (with config override)
+    ('L016', 'pass', 'SELECT COUNT(*) FROM tbl\n', None,
+     {'rules': {'max_line_length': 30}}),
+    # Check we move comments correctly
+    ('L016', 'fail', 'SELECT 1 -- Some Comment\n',
+     '-- Some Comment\nSELECT 1\n',
+     {'rules': {'max_line_length': 18}}),
+    # Check we can add newlines after dedents (with an indent)
+    ('L016', 'fail', '    SELECT COUNT(*) FROM tbl\n',
+     '    SELECT COUNT(*)\n    FROM tbl\n',
+     {'rules': {'max_line_length': 20}}),
+    # Check we handle indents nicely
+    ('L016', 'fail', 'SELECT 12345\n',
+     'SELECT\n    12345\n',
+     {'rules': {'max_line_length': 10}}),
+    # Check priority of fixes
+    ('L016', 'fail', 'SELECT COUNT(*) FROM tbl -- Some Comment\n',
+     '-- Some Comment\nSELECT COUNT(*)\nFROM tbl\n',
+     {'rules': {'max_line_length': 18}}),

```

Comment:
@barrywhart - I've done the first go at actually *fixing* some of the indentation issues. So far I've just done it via a new rule which checks line lengths. If the line is over a given length, it is then shortened in a variety of ways, which should be roughly in line with the guidelines we discussed before. The rules won't be exactly the same, because things were a bit trickier in practice than I hoped.

What do you think to these test cases? What would be really helpful is if you could battle test this rule a little, perhaps on your existing codebase and see how it runs. I know there will be plenty of bugs at this stage, but I'm curious which ones come up first.

I might next try and actually *lint* indentation changes between lines using some of the same functionality.

Reply:
Looks like a good start! I'm beginning to test this on our own SQL now. I'll follow up with any issues I find.

Reply:
@alanmcruickshank: I had a few spare hours and created several tickets a day or two ago -- hope that helps!

Most of the SQL I was testing on had already been cleaned up with respect to style and line-length, so I found some crash bugs but didn't see any actual changes to the SQL other than comments. Is there a way to specify the `max_line_length` when running `sqlfluff fix` from the command line, either as a command-line parameter or in `.sqlfluff`? If so, that will be very helpful for my testing, as I can do more thorough testing with my existing SQL, e.g. to target a shorter line length.

Reply:
I've raised a new issue for doing it from the command line #135 

It's definitely configurable in `.sqlfluff` already, but I totally expect that command line might be more helpful.